x-pack/test/functional_with_es_ssl/apps/triggers_actions_ui/alerts_list.ts

```diff
@@ -640,24 +639,29 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       // Select enabled
       await testSubjects.click('ruleStatusFilterButton');
       await testSubjects.click('ruleStatusFilterOption-enabled');
+      await new Promise((res) => setTimeout(res, 1000));
       await assertRulesLength(1);
 
       // Select disabled
       await testSubjects.click('ruleStatusFilterOption-enabled');
       await testSubjects.click('ruleStatusFilterOption-disabled');
+      await new Promise((res) => setTimeout(res, 1000));
```

Comment:
Is there anything on the page you can check for before asserting the rules length? Using a sleep like this is only going to work in very specific timing conditions which change constantly.

If there's a loading message in the UI that tells users the list is updating, then we should use the same indication to tell the tests that the list is still loading, like:
```
await testSubjects.waitForDeleted('ruleTableLoading');
await assertRulesLength(1);
```

Reply:
Yup, checkout my `wait` functions [in this test](https://github.com/elastic/kibana/blob/main/x-pack/test/functional_with_es_ssl/apps/triggers_actions_ui/alerts_table.ts#L133)